### ISSUE TITLE
Incomplete parser migration of Arrow library

### ISSUE THREAD

**Issue by corradio** [tmrowco/electricitymap-contrib#2470]
From what I see there's quite a few places where `.replace(` is used instead of `.shift(` on an arrow object?
If it is confirmed that a lot of parsers fail, then we could do a global find/replace.
cc @systemcatch 

Here's the output of the `ag` command:

src/electricitymap❯ ag 'replace\(h' contrib
contrib/parsers/MX.py
55:    dt = dt.replace(hour=int(hour) - 1, tzinfo=tzoffset)

contrib/parsers/CA_AB.py
130:                'datetime': arrow.get(rowIndex, 'MM/DD/YYYY').replace(hours=hours, tzinfo=ab_timezone).datetime,

contrib/parsers/US_CA.py
71:        date = arrow.utcnow().to('US/Pacific').replace(hour=h, minute=m,
224:        date = arrow.utcnow().to('US/Pacific').replace(hour=h, minute=m,

contrib/parsers/CA_ON.py
77:    dt = arrow.get(target_datetime).to(tz_obj).replace(hour=0, minute=0, second=0, microsecond=0)
287:        return dt.replace(hour=hour, minute=minute).datetime
349:    print('data should be for {}'.format(now.replace(hour=2).to(tz_obj).format('YYYY-MM-DD')))
350:    print('fetch_production("CA-ON", target_datetime=now.replace(hour=2)) ->')
351:    print(fetch_production("CA-ON", target_datetime=now.replace(hour=2)))
365:    print('data should be for {}'.format(now.replace(hour=2).to(tz_obj).format('YYYY-MM-DD')))
366:    print('fetch_price(target_datetime=now.replace(hour=2)) ->')
367:    print(fetch_price(target_datetime=now.replace(hour=2)))
388:    print('data should be for {}'.format(now.replace(hour=2).to(tz_obj).format('YYYY-MM-DD')))
389:    print('fetch_exchange("CA-ON", "CA-QC", target_datetime=now.replace(hour=2)) ->')
390:    print(fetch_exchange("CA-ON", "CA-QC", target_datetime=now.replace(hour=2)))

contrib/parsers/US_PJM.py
84:    arr_dt = arrow.now('America/New_York').replace(hour=hour, minute=minute)
294:        arr_dt = arr_date.replace(hour=arr_time.hour, minute=arr_time.minute).datetime

contrib/parsers/FR.py
174:                datetime = start_date.replace(hours=+period).datetime

contrib/parsers/SV.py
126:        dt = (at.replace(hour=int(hour), minute=0, second=0)).datetime

contrib/parsers/CR.py
145:        data_time = day.replace(hour=hour, minute=0, second=0, microsecond=0).datetime

contrib/parsers/lib/IN.py
25:    india_date_time = india_now.replace(hour=time.hour, minute=time.minute, second=time.second)

contrib/parsers/RU.py
223:        dt = today.replace(hour=hour).floor('hour').datetime

contrib/parsers/NI.py
324:    midnight_local_time = arrow.utcnow().to(TIMEZONE).replace(hour=0, minute=0, second=0,
339:        price_date = midnight_local_time.replace(hour=hour)

contrib/parsers/AR.py
681:    datetime = (at.replace(hour=int(find_hour), minute=0, second=0)).datetime

contrib/parsers/CY.py
110:        res[arrow.now('Europe/Nicosia').floor('day').replace(hour=hour, minute=minute)] = {
120:        res2[arrow.now('Europe/Nicosia').floor('day').replace(hour=hour, minute=minute)] = {

contrib/parsers/TR.py
161:                    data['datetime'] = tr_datetime.replace(hour=int(datapoint['saat'])).datetime

contrib/parsers/BD.py
34:        dt_aware = target_datetime.replace(hour=int(hour), minute=int(minute), tzinfo='Asia/Dhaka')

**Comment by systemcatch** [comment]
Hey Oli, let me double check but most of those examples should use replace.

**Comment by systemcatch** [comment]
Yeah only the FR example was wrong (that's been fixed in #2451). The others are using `replace` correctly, below is an example of the difference between the two methods.

```shell
(EM) chris@ThinkPad:~/electricitymap$ python
Python 3.7.4 (default, Sep 19 2019, 11:01:37) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> dt=arrow.utcnow()
>>> dt
<Arrow [2020-05-04T21:11:03.827374+00:00]>
>>> dt.replace(hour=10)
<Arrow [2020-05-04T10:11:03.827374+00:00]>
>>> dt.shift(hours=+10)
<Arrow [2020-05-05T07:11:03.827374+00:00]>
```

**Comment by corradio** [comment]
got it! Thanks for clarifying.
